src/job.rs

```diff
@@ -1,10 +1,24 @@
-extern chrono;
+extern crate chrono;
 
 use chrono::prelude::*;
 
-mod job {
-    struct Job<T> {
-        activity: Fn<T>,
-        deadline: DateTime,
+#[derive(Debug)]
+struct Activity {
+    command: String,
+    args: Vec<String>,
+}
+
+#[derive(Debug)]
+pub struct Job {
+    activity: Option<Activity>,
+    deadline: Option<DateTime<Utc>>,
```

Comment:
Does it make sense to ever construct a job that has no deadline, even if that deadline is "right now"? `None` could encode "right now" but it might be better as a clearer type, such as:

```
enum ExecutionTy {
  Immediate,
  Scheduled(DateTime<Utc>),
}
```